Feature - Product tag editing

lguichard opened the pull request:
Closes #1373

lguichard committed:
Implement tags on edit product

wychoong commented on packages/admin/resources/views/livewire/components/tags.blade.php:
```diff
@@ -1,14 +1,17 @@
                 this.tags.push(trimTag);
             }            
+
+            console.log(trimTag);
```
please cleanup the troubleshoot code

lguichard replied:
Oups ! Thx

wychoong commented on packages/admin/resources/views/livewire/components/tags.blade.php:
```diff
@@ -1,14 +1,17 @@
     x-data="{
         tags: $wire.entangle('tags'),
-        value: $wire.entangle('searchTerm', true),
+        newValue: $wire.entangle('searchTerm', true),
```
don't think this change is necessary as it's not really a new value either

lguichard replied:
I must be change the name of variable, there are a conflict with keyup.enter on input "$event.target.value" was empty.

That fixed two small bug on components: tag empty and key.enter event

wychoong commented on packages/admin/src/Filament/Resources/ProductResource.php:
```diff
@@ -175,9 +176,8 @@ protected static function getStatusFormComponent(): Component
-            ->label(__('lunarpanel::product.form.tags.label'))
-            ->placeholder('TODO: convert to Spatie Tags');
+        return Tags::make('tags')
+            ->label(__('lunarpanel::product.form.tags.label')); // TODO: convert to Spatie Tags
```
remove the comment

lguichard replied:
Why ?

wychoong replied:
because the text was a placeholder for a feature to be implemented.
the comment should be consider as outdated as mentioned in #1373 not to use spatie's for now. When the time comes to change to spatie's, that would be another issue/PR

wychoong reviewed (commented):
can you also provide a screenshot of the form component with label 😄

lguichard committed:
Cleaning

lguichard commented:
> can you also provide a screenshot of the form component with label 😄

It's same that order tags editin
![Capture d’écran 2023-12-26 à 16 24 08](https://github.com/lunarphp/lunar/assets/6489718/255cf938-c574-4737-9185-5795efbdb3fd)
g.

wychoong commented:
> > can you also provide a screenshot of the form component with label 😄
> 
> It's same that order tags editin ![Capture d’écran 2023-12-26 à 16 24 08](https://private-user-images.githubusercontent.com/6489718/292888096-255cf938-c574-4737-9185-5795efbdb3fd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM2MDQ3MjYsIm5iZiI6MTcwMzYwNDQyNiwicGF0aCI6Ii82NDg5NzE4LzI5Mjg4ODA5Ni0yNTVjZjkzOC1jNTc0LTQ3MzctOTE4NS01Nzk1ZWZiZGIzZmQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMjZUMTUyNzA2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZmJjODUwNzdmN2U1NjBmZmIzZTc3ZDYxYmM1NGRhYzFhZDBiZTY2OTJhZGRkMTYyNmRkZDM3MzQ3OTI2NDk5ZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.DembWU5jfsJLv-PE2k-X-MZRih7VKSiIMt4ELB6glF4) g.

need some tweaking for the label, and ideally use [filament's wrapper/blade](https://filamentphp.com/docs/3.x/support/blade-components/input-wrapper) instead of managing the styling ourselves

maybe try fieldset
```html
<x-filament::fieldset
      :label="$getLabel()"
>
      @livewire('lunar.admin.livewire.components.tags', [
          'taggable' => $getRecord()
      ])
</x-filament::fieldset>
```

also the save button for the tags will be confusing 🤔

lguichard commented:
The save button is a behavior choice by core team, i'm also confusing. 

 I had tested to change getDefaultForm but I wasn't really satisfied. Move to sub navigation would be the right solution

lguichard committed:
Cleaning

lguichard committed:
Revert waiting feedback

glennjacobs commented:
As the tags field is part of the main form, it shouldn't have a separate "Save" button. It should save along with the other components.

lguichard committed:
Refactor extends TagsInput default Filament component

lguichard committed:
Remove custom tags livewire component on OrderResource & cleaning old component

lguichard commented:
@glennjacobs I refactored the code. I replaced and removed the custom Livewire component. Now we use the native filament tag input override. When we convert the tags to Spatie the change will be very fast.

Now the panel have only two class that manage Tag resource

``` php

// Edit form tags
use Lunar\Admin\Support\Forms\Components\Tags inherit from Filament\Forms\Components\TagsInput

// Usage
Tags::make('tags')
  ->suggestions(Tag::all()->pluck('value')->all())
  // ....
  ->label('My label');
``` 

``` php
// Display list tags
Lunar\Admin\Support\Infolists\Components\Tags inherit from Filament\Infolists\Components\TextEntry; 

// Usage
Tags::make('tags')
  ->label('My Label');
```

The form in the order edit is same that behavior Shipping Address and Billing Address.

<img width="836" alt="Capture d’écran 2023-12-28 à 21 04 54" src="https://github.com/lunarphp/lunar/assets/6489718/756cf209-8444-451d-b776-857fe7a1f9f1">

<img width="424" alt="Capture d’écran 2023-12-28 à 20 52 06" src="https://github.com/lunarphp/lunar/assets/6489718/56ed57c0-d13a-4b1f-a856-35278023259d">

lguichard committed:
Cleaning

lguichard committed:
Coding style

glennjacobs committed:
Merge branch '1.x' into feature/1373

glennjacobs committed:
Merge branch '1.x' into feature/1373

glennjacobs committed:
Merge branch '1.x' into feature/1373

glennjacobs committed:
Merge branch '1.x' into feature/1373
